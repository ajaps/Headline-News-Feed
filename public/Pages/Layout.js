import firebase from 'firebase';
import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link,
  Redirect,
  Switch,
} from 'react-router-dom';

import Spinner from 'react-spinner';

import Dashboard from './Dashboard';
import Login from './Login';


firebase.initializeApp({
  apiKey: process.env.API_KEY,
  authDomain: process.env.AUTH_DOMAIN,
  databaseURL: process.env.DATABASE_URL,
  projectId: process.env.PROJECT_ID,
  storageBucket: process.env.STORAGE_BUCKET,
  messagingSenderId: process.env.MESSAGING_SENDER_ID,
});

export default class Layout extends React.Component {
  constructor() {
    super();
    this.state = { url: '/Dasboard' };
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged(user => {
      this.setState({ user });
    });
  }

  // Sign Out from Firebase/Application
  logout() {
    this.setState({ user: null });

    firebase.auth().signOut().then(() => {
      // Sign-out successful.
    }).catch((error) => {
      // An error happened.
      alert(error);
    });
  }

  // To change the URL
  changeUrl(url) {
    this.setState({ url });
  }

  // Sign In to Firebase/Application
  signIn() {
    const provider = new firebase.auth.GoogleAuthProvider();
    firebase.auth().signInWithPopup(provider).then((result) => {
    }).catch((error) => {
      alert(error.message);
    });
  }

  render() {
    const { user } = this.state;
    return (
      user === undefined ?
        <Spinner /> :
        <Router>
          <Switch>
            <Route exact path="/" component={() => user ? <Redirect to="/Dashboard" /> : <Login />} />
            <Route path="/Login" component={() => user ? <Redirect to="/Dashboard" /> : <Login logInFirebase={this.signIn.bind(this)} />} />
            <Route path="/Dashboard" component={() => user ? <Dashboard changeUrl={this.changeUrl.bind(this)} logout={this.logout.bind(this)} params={this.state} /> : <Redirect to="/Login" />} />
          </Switch>
        </Router>
    );
  }
}
