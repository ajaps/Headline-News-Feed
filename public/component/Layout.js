import React from 'react';

import Footer from './Footer';
import Header from './Header';


export default class Layout extends React.Component {
  constructor() {
    super();
    this.state = { title: 'Ajaps Franklin that BOSS!!' };
  }

  changeTitle(title) {
    this.setState({ title });
  }

  render() {
    /*
    setTimeout(() => {
      this.setState({ name: 'Tunde Abiola' });
    }, 6000);
    */
    return <div>
      <Header changeTitle={this.changeTitle.bind(this)} title={this.state.title} />
      <h4>Hello from {this.state.title}</h4>
      <Footer />
    </div>;
  }
}
