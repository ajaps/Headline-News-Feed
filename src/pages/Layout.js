import { createBrowserHistory } from 'history';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import React from 'react';
import ReactLoading from 'react-loading';

import Dashboard from './Dashboard';
import firebaseAuth from '../config/firebase';
import Footer from '../components/Footer.jsx';
import Header from '../components/Header.jsx';
import Login from './Login';
import Page404 from './Page404.jsx';
import { PrivateRoute } from '../components/PrivateRoute';
import { PublicRoute } from '../components/PublicRoute';


const history = createBrowserHistory();


class Layout extends React.Component {
  constructor() {
    super();
    this.state = { authenticated: false, loading: true, };
  }

  componentDidMount() {
    this.removeListener = firebaseAuth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({ authenticated: true, loading: false, });
      } else {
        this.setState({ authenticated: false, loading: false, });
      }
    });
  }

  componentWillUnmount() {
    this.removeListener();
  }

  render() {
    return this.state.loading === true ?
    <ReactLoading className="spinner" width="150px"
    type="spin" color="#3b5998"/> :
    (
      <div>
        <Header containLogoutBtn={this.state.authenticated} />
        <Router history={history}>
          <Switch>
            <PublicRoute authenticated={this.state.authenticated}
            exact path="/" component={Login} />

            <PublicRoute authenticated={this.state.authenticated}
            path="/login" component={Login} />

            <PrivateRoute authenticated={this.state.authenticated}
            path="/dashboard" component={Dashboard} />

            <Route path="*" component={Page404} />
          </Switch>
        </Router>
        <Footer />
      </div>
    );
  }
}

export default Layout;
