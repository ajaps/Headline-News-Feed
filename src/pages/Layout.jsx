import { createBrowserHistory } from 'history';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import React from 'react';
import ReactLoading from 'react-loading';

import Dashboard from './Dashboard.jsx';
import firebaseAuth from '../config/firebase';
import Footer from '../components/Footer.jsx';
import Header from '../components/Header.jsx';
import Login from './Login.jsx';
import PageNotFound from './PageNotFound.jsx';
import { PrivateRoute } from '../components/PrivateRoute';
import { PublicRoute } from '../components/PublicRoute';


const history = createBrowserHistory();


class Layout extends React.Component {
  constructor() {
    super();
    this.state = { authenticated: false, loading: true, };
  }

  componentDidMount() {
    this.removeListener = firebaseAuth().onAuthStateChanged((user) => {
      if (user) {
        const name = user.displayName;
        this.setState({ authenticated: true, loading: false, user: name });
      } else {
        this.setState({ authenticated: false, loading: false, });
      }
    });
  }

  componentWillUnmount() {
    this.removeListener();
  }

  render() {
    const name = this.state.user;
    return this.state.loading === true ?
    <ReactLoading
        className="spinner" color="#3b5998" type="spin" width="150px"
    /> :
    (
      <div>
        <Header containLogoutBtn={this.state.authenticated} user={name} />
        <section>
        <Router history={history}>
          <Switch>
            <PublicRoute
                authenticated={this.state.authenticated}
                component={Login} exact path="/"
            />

            <PublicRoute
                authenticated={this.state.authenticated}
                component={Login} path="/login"
            />

            <PrivateRoute
                authenticated={this.state.authenticated}
                component={Dashboard} exact path="/dashboard"
            />

            <Route component={PageNotFound} path="*" />
          </Switch>
        </Router>
        </section>
        <Footer />
      </div>
    );
  }
}

export default Layout;
