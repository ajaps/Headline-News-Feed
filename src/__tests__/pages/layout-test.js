import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Layout from '../../pages/Layout';

jest.mock('../../config/firebase', () => ({
}));

describe('Layout page', () => {
  let app;
  beforeEach(() => {
    app = shallow(<Layout />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should render only loading element at initial page load', () => {
    expect(app.find('Loading').length).toBe(1);
    expect(app.find('div').length).toBe(0);
  });
});
