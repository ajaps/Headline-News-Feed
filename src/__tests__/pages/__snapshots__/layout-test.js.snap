// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout page should render as expected 1`] = `<h1 />`;
